src/test/client/nonblocking.cc

```diff
@@ -716,3 +718,76 @@ TEST_F(TestClient, LlreadvContiguousLlwritevNonContiguous) {
   client->ll_release(fh);
   ASSERT_EQ(0, client->ll_unlink(root, filename, myperm));        
 }
+
+TEST_F(TestClient, LlreadvLlwritevLargeBuffers) {
+  /* Test that async I/O code paths handle large buffers (total len >= 4GiB)*/
+  int mypid = getpid();
+  char filename[256];
+
+  client->unmount();
+  TearDown();
+  SetUp();
+
+  sprintf(filename, "test_llreadvllwritevlargebuffers%u", mypid);
+
+  Inode *root, *file;
+  root = client->get_root();
+  ASSERT_NE(root, (Inode *)NULL);
+
+  Fh *fh;
+  struct ceph_statx stx;
+
+  ASSERT_EQ(0, client->ll_createx(root, filename, 0666,
+                                  O_RDWR | O_CREAT | O_TRUNC,
+                                          &file, &fh, &stx, 0, 0, myperm));
+
+  struct statvfs stbuf;
+  int64_t rc;
+  rc = client->ll_statfs(root, &stbuf, myperm);
+  ASSERT_EQ(rc, 0);
+  int64_t fs_available_space = stbuf.f_bfree * stbuf.f_bsize;
+  ASSERT_GT(fs_available_space, 0);
+
+  const int64_t BUFSIZE = INT_MAX + 1;
+  int64_t bytes_written = 0, bytes_read = 0;
+
+  std::unique_ptr<C_SaferCond> writefinish = nullptr;
```

Comment:
nit: why not jus use `C_SaferCond writefinish`?

(I think I left a similar comment in one of your other PRs)

Reply:
This is coming from the time when I was working on my first ever async io related tracker where I was following the existing test case which came from frank's PR. If you see then the entire test file https://github.com/ceph/ceph/blob/main/src/test/client/nonblocking.cc is making of `std::unique_ptr<C_SaferCond> writefinish = nullptr;`. I can change this right now or I can refactor all these calls altogether after all of my PRs are merged.

Reply:
Oh, looks like I have a cleanup tracker for async io test cases - https://tracker.ceph.com/issues/66823. I can add this task to it and later cleanup everything together?

Reply:
Let's cleanup this change to use C_SaferCond directly.

Reply:
done https://github.com/ceph/ceph/compare/7a98db92b222481dac73eb7839a51d83fd79a675..d4cbde0d7d105580c29e0ebfe6f4d4b37bafdf16